### PR TITLE
Jerry nes rebased

### PR DESCRIPTION
**Pull request by JerryIshihara**

### CODE REVIEW

File: src/smoke.hh [hunk 1]
```diff
@@ -1,16 +1,39 @@
...
+    int width;
+    int height;
+    int depth;
+    int grid_width;
+    int grid_height;
+    int grid_depth;
+
+    void generateParticles(Emittor emittor, int num_particles);
     void update(double delta_t);
+    void build_spatial_map();
```
Comment (geos98) [inline src/smoke.hh:36]: use `const Emitter& emitter`
Try pass by reference when ever possible for classes

File: src/navier_stoke_solver.hh [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    // nanogui::Vector3f velocity;
+    // nanogui::Vector3f forces;
+    ~NavierStokeSolver(){};
+
+    // TODO: maybe change this to adjustable
+    double base_density;
+    double fluid_stiffness;
+    nanogui::Vector3f gravity = nanogui::Vector3f(0.0f, -9.81f, 0.0f);
+
+    void update(std::vector<Particle *> grid_particles, double delta_t);
```
Comment (geos98) [inline src/navier_stoke_solver.hh:54]: Use `const std::vector<Particle *>&` 
We don't need to pass the vector by value

File: src/navier_stoke_solver.hh [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    void simplified_update(std::vector<Particle *> grid_particles, double delta_t);
```
Comment (geos98) [inline src/navier_stoke_solver.hh:55]: Same as above

File: src/smoke_simulator.cc [hunk 4]
```diff
@@ -18,7 +18,7 @@ SmokeSimulator::SmokeSimulator(string shader_dir, nanogui::Screen *screen, GLFWw
     Vector3f init_velocity = Vector3f(0, 0.5, 0);
     Vector4f init_color = Vector4f(66 / 256.0f, 135 / 256.0f, 245 / 256.0f, 1.0f);
     emittor = Emittor(Vector3f(0, -1, 0), init_velocity, init_color);
-    smoke = Smoke();
+    smoke = Smoke(400, 400, 400, 1000, 1000, 1000);
```
Comment (geos98) [inline src/smoke_simulator.cc:21]: Unless I am missing something, there is no such constructor for Smoke

**Review by geos98 — CHANGES_REQUESTED** [review]

File: src/smoke_simulator.cc (hunk 4 above)
Comment (JerryIshihara) [reply src/smoke_simulator.cc:21]: updated header

**Pull request merged**